src/supervisor/workload-reader.ts

```diff
@@ -16,7 +17,8 @@ const deploymentReader: IWorkloadReaderFunc = async (workloadName, namespace) =>
 
   if (!deployment.metadata || !deployment.spec || !deployment.spec.template.metadata ||
       !deployment.spec.template.spec || !deployment.status) {
-    // TODO(ivanstanev): add logging to know when/if it happens!
+    logger.info({workloadName, namespace}, 'kubernetes api could not return workload');
```

Comment:
Just realised after the second review: this line seems to be repeating lots of times with identical structure, worth putting into a function? Something like `logIncompleteWorkload(workloadName, namespace);`, name up to you 🙏